crates/gateway-client/src/lib.rs

```diff
@@ -213,10 +223,19 @@ impl Client {
 impl GatewayApi for Client {
     #[tracing::instrument(skip(self))]
     async fn block(&self, block: BlockId) -> Result<reply::MaybePendingBlock, SequencerError> {
+        self.block_with_retry(block, Self::RETRY).await
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn block_with_retry(
```

Comment:
Why leak and bake retry into the API? Keeping it opaque is much simpler/cleaner/testable way IMHO.